src/components/vuestic-components/va-infinite-scroll/VaInfiniteScroll.vue

```diff
@@ -0,0 +1,226 @@
+        <va-progress-circle
+          size="small"
+          :thickness="0.15"
+          :color="error ? $themes.danger : $themes.primary"
+          indeterminate
+        />
+      </div>
+    </slot>
+  </component>
+</template>
+
+<script>
+import * as _ from 'lodash'
+import VaProgressCircle from '../va-progress-bar/progress-types/VaProgressCircle'
+import { makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import { sleep } from '../../../services/utils'
+
+export default {
+  name: 'VaInfiniteScroll',
+  components: { VaProgressCircle },
+  mixins: [
+    makeContextablePropsMixin({
+      offset: {
+        type: Number,
+        default: 1,
+      },
+      reverse: {
+        type: Boolean,
+        default: false,
+      },
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      scrollTarget: {
+        type: [Element, String],
+        default: null,
+      },
+      debounce: {
+        type: Number,
+        default: 100,
+      },
+      load: {
+        type: Function,
+        default: () => {},
+      },
+      tag: {
+        type: String,
+        default: 'div',
+      },
+    }),
+  ],
+  data () {
+    return {
+      index: 0,
+      fetching: false,
+      error: false,
+      initialHeight: null,
+    }
+  },
+  watch: {
+    disabled (value) {
+      if (value) {
+        this.stop()
+      } else {
+        this.resume()
+      }
+    },
+    debounce (value) {
+      this.setDebounce(value)
+    },
+  },
+  methods: {
+    onLoad () {
+      if (this.disabled || this.error || this.fetching) {
+        return
+      }
+        
+      const { scrollTop, scrollHeight } = this.scrollTargetElement
+      const containerHeight = this.scrollTargetElement.offsetHeight
+      const isLoadingRequired = this.reverse
+        ? scrollTop < this.scrollAmount
+        : scrollTop + containerHeight + this.scrollAmount >= scrollHeight;
+      if (!isLoadingRequired) return
+
+      this.fetching = true
+      this.scrollTop = this.reverse ? 0 : this.$el.offsetHeight
+      this.initialHeight = this.$el.offsetHeight
+
+      this.load()
+        .then(this.finishLoading)
+        .catch(this.onError)
+    },
+    onError (e) {
+      this.stop()
+      this.error = true
+      this.fetching = true
+      sleep(2000).then(this.stopErrorDisplay).then(this.resume)
+    },
+    stopErrorDisplay () {
+        this.scrollTargetElement.scrollTop -= this.reverse
+          ? this.scrollTargetElement.scrollTop - this.scrollAmount
+          : this.scrollAmount
+        this.error = false
+        this.fetching = false
+    },
+    finishLoading () {
+      this.fetching = false
+      if (this.reverse) {
+        const heightDifference =
+        this.$el.offsetHeight - this.initialHeight
+        this.scrollTargetElement.scrollTop = heightDifference
+      }
+    },
+    resume () {
+      if (!this.disabled) {
+
+          this.scrollTargetElement.addEventListener(
```

Comment:
run lint + double check why it doesn't work for you. You might want to merge from dev for updated deps.